packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaForm.tsx

```diff
@@ -501,6 +504,33 @@ function TableColumnSchemaFormRow(props: TableColumnSchemaFormRowProps) {
     ),
   )
 
+  // Find the closest match between passed column model and current column model with name and type
+  const findClosestMatch = (
+    columnModels: ColumnModel[],
+  ): ColumnModel | undefined => {
+    let closestMatch: ColumnModel | undefined
+
+    for (const model of columnModels) {
+      if (isCloseMatch(model)) {
+        closestMatch = model
+        break
+      }
+    }
+
+    return closestMatch
+  }
+
+  const isCloseMatch = (model: ColumnModel): boolean => {
+    return (
+      model.name === columnModel.name &&
+      model.columnType === columnModel.columnType
+    )
+  }
```

Comment:
Recommend pulling these out of the component, rename to `findMatchingColumnModel` just because `findClosestMatch` is a little vague